### PR TITLE
Add socket check before trainer startup

### PR DESCRIPTION
**Pull request by velconia**

### CODE REVIEW

File: resnet50_paddle_cloud_dist/fluid/fluid_benchmark.py [hunk 1]
```diff
@@ -360,6 +374,13 @@ def main():
                 "Must configure correct environments to run dist train.")
         train_args.extend([train_prog, startup_prog])
         if args.gpus > 1 and os.getenv("TRAINING_ROLE") == "TRAINER":
+            max_count = 100
+            time.sleep(60 * 2)
```
Comment (panyx0718) [inline resnet50_paddle_cloud_dist/fluid/fluid_benchmark.py]: no need for this?

Comment (velconia) [reply resnet50_paddle_cloud_dist/fluid/fluid_benchmark.py]: aligned with QA, ```sleep``` is needed for ```PaddleCloud```

Comment (panyx0718) [reply resnet50_paddle_cloud_dist/fluid/fluid_benchmark.py]: but you have retry below? Maybe it doesn't need 120 sec?

Comment (velconia) [reply resnet50_paddle_cloud_dist/fluid/fluid_benchmark.py]: we should wait at least 120 secs ...

Comment (panyx0718) [reply resnet50_paddle_cloud_dist/fluid/fluid_benchmark.py]: if you retry every 60 secs. And retry 10 times, it would be 600secs?...

**Pull request closed without merging**